src/sets.jl

```diff
@@ -807,6 +807,123 @@ end
 
 dimension(set::Complements) = 2 * set.dimension
 
+"""
+    AllDifferent(dimension::Int)
+
+The set corresponding to an all-different constraint.
+
+All expressions of a vector-valued function are enforced to take distinct
+values in the solution: for all pairs of expressions, their values must
+differ. This constraint is only defined for integer-valued expressions.
+
+This constraint is sometimes called `distinct`.
+"""
+struct AllDifferent <: AbstractVectorSet
+    dimension::Int
+end
+
+"""
+    Domain{T <: Int}(values::Set{T})
+
+The set corresponding to an enumeration of constant values.
+
+The value of a scalar function is enforced to take a value from this set of
+values.
+
+This constraint is sometimes called `in`.
+"""
+struct Domain{T <: Int} <: AbstractScalarSet
```

Comment:
Again, why restrict to `Int`? Also, `Int` here is a concrete type, so `T <: Int` is just `Int`.